include/tvm/script/ir_builder/tir/ir.h

```diff
@@ -428,16 +428,26 @@ void Evaluate(PrimExpr value);
 
 /*!
  * \brief Create a TIR var that represents a pointer
+ *
  * \param dtype The data type of the pointer.
+ *
  * \param storage_scope The storage scope of the pointer.
+ *
  * \param is_size_var Whether the pointer is a size var.
+ *
+ * \param is_unknown_type Used to distinguish between
+ * `PrimType(DataType::Handle())` and
+ * `PointerType(PrimType(DataType::Void()))`.  If true, resolve dtype
+ * of `Void()` as `PrimType`, and if false resolve dtype of `Void()`
```

Comment:
`Void` is defined as `DataType(kHandle, 0, 0)`, while a legitimate handle has non-zero bits and lanes.  Do we need an additional parameter here instead of passing the right thing in `dtype`?

Reply:
I believe so, because the `dtype` is providing the type of the pointed-to object, not the type of the pointer itself.  While we could special-case passing of a `DataType(kHandle, 0, 0)` to produce `PrimType(DataType::Handle())`, that would give unexpected results if somebody tries to write a pointer-to-pointer as `T.handle("handle")` and accidentally hits the special-case.